kernel/Kconfig

```diff
@@ -195,6 +195,13 @@ config APPLICATION_MEMORY
 	  will have the project-level application objects and any libraries
 	  including the C library in it.
 
+config RAM_FUNCTION
```

Comment:
- I would rather name this option as XIP_EXECUTE_FROM_RAM, instead, to reflect what we are actually enabling (or something similar)
- @andrewboie  Do we need to have a dependence on XIP? 
- Also, this is not, really, a Kernel option - rather a generic arch option, right? If you agree, I suggest to move the option to arch/Kconfig

Reply:
Current __ramfunc only support ARM platform. maybe arm/Kconfig is a better place.

Reply:
> XIP_EXECUTE_FROM_RAM
I prefer CONFIG_RAM_FUNCTION because the sections's name is ramfunc.
Also we can discussed more about this

Reply:
> Also, this is not, really, a Kernel option - rather a generic arch option, right? If you agree, I suggest to move the option to arch/Kconfig

Does the others SoC need ramfunc? if so I think should leave this to the kernel/Kconfig.

Reply:
> others

Allowing the firmware to execute code from SRAM has little or nothing to do with the Zephyr Kernel. That's why I suggested the option to be defined in arch/ instead.

Reply:
> I prefer CONFIG_RAM_FUNCTION because the sections's name is ramfunc.

What you are actually contributing here, is a way for the firmware to execute functions located in RAM. We need to propose a name that reflects exactly this thing.